ffyaml/ffyaml_test.go

```diff
@@ -71,3 +71,75 @@ func TestParser(t *testing.T) {
 		})
 	}
 }
+
+func TestParserAllTypesEmptyVals(t *testing.T) {
+	t.Parallel()
+
+	for _, testcase := range []struct {
+		name string
+		file string
+		want fftest.Vars
+	}{
+		{
+			name: "basic empty vals",
+			file: "testdata/empty_basic_vals.yaml",
+			want: fftest.Vars{S: "", I: 0, F: 0.00, D: 0 * time.Second, B: false, X: nil},
+		},
+	} {
+		t.Run(testcase.name, func(t *testing.T) {
+			fs, testVars := fftest.EmptyValsFS()
+			testVars.ParseError = ff.Parse(fs, []string{},
+				ff.WithConfigFile(testcase.file),
+				ff.WithConfigFileParser(ffyaml.Parser),
+			)
+			pError := testVars.ParseError
+			checkParseErr(t, pError)
+
+			if err := fftest.Compare(&testcase.want, testVars); err != nil {
+				t.Fatal(err)
+			}
+		})
+	}
+}
+
+func checkParseErr(t *testing.T, pError error) {
+	if pError != nil {
+		var errMsg string
+		errMsg = "ParseError: " + pError.Error()
+		if errors.Is(pError, ff.StringConversionError{}) {
+			errMsg = errMsg + ";errType: " + "StringConversionError"
+		}
+		t.Fatal(errMsg)
+	}
+}
```

Comment:
Nope 😉 Do it properly at the callsites.

```suggestion
```

Reply:
Sorry, I don't understand what you mean by this: 

> Do it properly at the callsites.

Reply:
Do not define or use a `checkParseErr` helper function. Check the error directly at the callsites.

Reply:
Done. I have reversed the refactoring of Checking the error logic, "inline'ing" the previous checkParseErr() functionality.